src/diagonal.works/b6/api/functions/collections.go

```diff
@@ -341,3 +342,155 @@ var _ api.Collection = &flattenCollection{}
 func flatten(_ *api.Context, c api.Collection) (api.Collection, error) {
 	return &flattenCollection{c: c}, nil
 }
+
+const maxBuckets = 5
+
+type bucket struct {
+	start, end interface{}
```

Comment:
For numeric buckets, isn't the start of bucket n equal to the (end of bucket n-1) - epsilon? ie doesn't it make more sense to have a bucket defined a being buckets[n-1] <= x < buckets[n] or similar, rather than redundantly storing both values - and that naturally means categorical buckets don't ignore one of the fields.

Reply:
so in this case where i m doing uniform based on data the bucket ranges do overlap, but they don't necessarily have to? 1. i wanted to give the user the option to "easily" hard code arbitrary bounds, and while it does make things slightly more redundant here 2. i think conceptually it makes more sense to have upper and lower bound 

wdyt

Reply:
What does it mean for a histogram to have buckets that overlap? ie that two bars next to each other each include a particular data point? That sounds broken, but maybe I'm missing something? Or did you just intend the function to be a more general form of bucketing that just for histograms?

Reply:
by overlap i meant touch / share bound, and i meant this let's us have buckets that *don't* share the bound, so like i can ask how many data points r in `1-100` and `1000-10000` buckets for instance, maybe that's silly?